Source/WebCore/css/ComputedStyleExtractor.cpp

```diff
@@ -2342,10 +2333,10 @@ static bool isLayoutDependent(CSSPropertyID propertyID, const RenderStyle* style
             return false;
         return !(style && style->marginTop().isFixed() && style->marginRight().isFixed()
             && style->marginBottom().isFixed() && style->marginLeft().isFixed())
-            || (isFlexItem(renderer) && rendererContainingBlockHasMarginTrim(downcast<RenderBox>(*renderer), { }));
+            || canCheckForTrimmedMargin(downcast<RenderBox>(*renderer), { });
```

Comment:
canCheckForTrimmedMargin is a somewhat confusing name in this context.
```
bool isLayoutDependent()
{
    if (canCheckForFoo())
        return hasFoo();
...
```
is what I would expect as opposed to
```
bool isLayoutDependent()
{
    return canCheckForFoo();
}
```

Reply:
I went ahead and moved this function outside of `ComputedStyleExtractor` and  into `RenderBox`. I also moved the responsibility of checking this away from the caller and into `hasTrimmedMargin` itself so that we can avoid pairing these function calls together at each call site. Let me know what you think